staging_vespalib/src/vespa/vespalib/util/singleexecutor.cpp

```diff
@@ -69,10 +97,8 @@ void
 SingleExecutor::run() {
     while (!_thread.stopped()) {
         drain_tasks();
-        _wakeupConsumerAt.store(_wp.load(std::memory_order_relaxed) + (_taskLimit.load(std::memory_order_relaxed) >> 2), std::memory_order_relaxed);
-        MonitorGuard guard(_consumerMonitor);
-        guard.wait(10ms);
-        _wakeupConsumerAt.store(0, std::memory_order_relaxed);
+        wakeupProducer();
```

Comment:
note that this signal may be lost since no state is being passed with the signal (the other thread might call wait after the signal has been discarded since no-one was waiting).

Reply:
Yes, correct. The purpose is to wake up one that is sleeping waiting for low watermark to be reached.